packages/nx/src/nx-init/add-nx-to-nest.ts

```diff
@@ -0,0 +1,339 @@
+    return;
+  }
+
+  const nestCacheableScripts = ['build', 'lint'];
+  const nestIgnoreScripts = ['start', 'start:dev', 'start:debug'];
+
+  const scripts = Object.keys(packageJson.scripts).filter((s) => {
+    if (nestCacheableScripts.includes(s) || nestIgnoreScripts.includes(s)) {
+      return false;
+    }
+
+    return !s.startsWith('pre') && !s.startsWith('post');
+  });
+
+  let cacheableOperations: string[];
+  let scriptOutputs = {};
+  let useCloud: boolean;
+
+  if (parsedArgs.yes !== true) {
+    output.log({
+      title: `
+    We are handling these scripts manually:
+    - build, start, start:debug, lint are moved to project.json as targets
+    - start:dev is the same as "start"
+    `,
+    });
+    output.log({
+      title: `🧑‍🔧 Please answer the following questions about the scripts found in your package.json in order to generate task runner configuration`,
+    });
+    cacheableOperations = (
+      (await enquirer.prompt([
+        {
+          type: 'multiselect',
+          name: 'cacheableOperations',
+          message:
+            'Which of the following scripts are cacheable? (Produce the same output given the same input, e.g. build, test and lint usually are, serve and start are not)',
+          choices: scripts,
+        },
+      ])) as any
+    ).cacheableOperations;
+
+    for (const scriptName of cacheableOperations) {
+      scriptOutputs[scriptName] = (
+        await enquirer.prompt([
+          {
+            type: 'input',
+            name: scriptName,
+            message: `Does the "${scriptName}" script create any outputs? If not, leave blank, otherwise provide a path (e.g. dist, lib, build, coverage)`,
+          },
+        ])
+      )[scriptName];
+    }
+
+    useCloud = await askAboutNxCloud();
+  } else {
+    cacheableOperations = parsedArgs.cacheable
+      ? parsedArgs.cacheable.split(',')
+      : [];
+    useCloud = false;
+  }
+
+  createNxJsonFile(
+    repoRoot,
+    [],
+    [...cacheableOperations, ...nestCacheableScripts],
+    {},
+    packageJson.name
+  );
+
+  const pmc = getPackageManagerCommand();
+
+  addDepsToPackageJson(repoRoot, useCloud);
+  addNestPluginToPackageJson(repoRoot);
+  markRootPackageJsonAsNxProject(
+    repoRoot,
+    cacheableOperations,
+    scriptOutputs,
+    pmc
+  );
+
+  createProjectJson(repoRoot, packageJson.name, nestCLIOptions);
+  removeNestCLIConfig(repoRoot);
+
+  updatePackageJsonScripts(repoRoot);
+  updateTsConfig(repoRoot, nestCLIOptions.sourceRoot);
+
+  output.log({ title: `📦 Installing dependencies` });
+
+  runInstall(repoRoot);
+
+  if (useCloud) {
+    initCloud(repoRoot);
+  }
+
+  printFinalMessage();
+}
+
+function printFinalMessage() {
+  output.success({
+    title: `🎉 Done!`,
+    bodyLines: [
+      `- Enabled computation caching!`,
+      `- Learn more at https://nx.dev/recipes/adopting-nx/adding-to-monorepo`,
+    ],
+  });
+}
+
+function parseNestCLIOptions(): NestCLIOptions {
```

Comment:
Since we are in a nest-cli workspace, we should be able to require in this right? https://github.com/nestjs/nest-cli/blob/master/lib/configuration/nest-configuration.loader.ts

If so, this could be type safe and not have to worry as much about nulls.